src/data_access/FirebaseRoomsDataAccessObject.java

```diff
@@ -147,32 +160,34 @@ public void removeUserFromRoom(
     }
   }
 
-  @Override
-  public void addUserToRoom(
-      User currentUser, DisplayUser newUser, LoginUserDataAccessInterface userDAO, Room room) {
-    String idToken = userDAO.getAccessToken(currentUser.getEmail(), currentUser.getPassword());
-
+  private void addUserToRoomData(DisplayUser newUser, String idToken, String roomId) {
     String jsonBody = JSONObject.quote(newUser.getName());
     String encodedEmail =
         Base64.getEncoder().encodeToString(newUser.getEmail().toLowerCase().getBytes());
-    String url =
-        String.format(Constants.ROOM_USERS_URL, room.getUid(), encodedEmail) + "?auth=" + idToken;
+    String url = String.format(Constants.ROOM_USERS_URL, roomId, encodedEmail) + "?auth=" + idToken;
     RequestBody body = RequestBody.create(jsonBody, MediaType.parse("application/json"));
 
     Request request = new Request.Builder().url(url).put(body).build();
 
     try {
       Response response = client.newCall(request).execute();
-      if (response.isSuccessful()) {
-        addRoomToUserData(newUser, idToken, room);
-      } else {
+      if (!response.isSuccessful()) {
         throw new IOException();
       }
     } catch (IOException | JSONException e) {
       throw new RuntimeException("Unable to add user to room. Please try again.");
     }
   }
 
+  @Override
+  public void addUserToRoom(
+      User currentUser, DisplayUser newUser, LoginUserDataAccessInterface userDAO, Room room) {
+    String idToken = userDAO.getAccessToken(currentUser.getEmail(), currentUser.getPassword());
+
+    addUserToRoomData(newUser, idToken, room.getUid());
```

Comment:
Might be nicer to pass the whole room object here, but this is fine.

Reply:
The reason we don't pass the whole room object is that we only have the roomId in our propogateDisplayNameChange call, and loading the whole room would take significantly longer.

Reply:
So essentially `addUserToRoomData` is taking the least amount of data possible to make it easier for propogateDisplayNameChange` to call it